### PR TITLE
feat(react): add legacy guest flow to useUserReturns hook

### PR DESCRIPTION
**Pull request by andreecosta**
## Description

This adds the legacy flow to useUserReturns hook

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/react/src/returns/hooks/types/useUserReturns.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+import type {
+  Config,
+  GetUserReturnsLegacyData,
+  GetUserReturnsQuery,
+} from '@farfetch/blackout-client';
 
 export type UseUserReturnsOptions = {
   enableAutoFetch?: boolean;
   fetchConfig?: Config;
   fetchQuery?: GetUserReturnsQuery;
```
Comment (boliveira) [inline packages/react/src/returns/hooks/types/useUserReturns.ts:10]:
```suggestion
  fetchQuery?: GetUserReturnsQuery | GetUserReturnsLegacyQuery;
```

File: packages/react/src/returns/hooks/useUserReturns.ts [hunk 2]
```diff
@@ -167,15 +175,38 @@ function useUserReturns(options: UseUserReturnsOptions = {}) {
...
 
+    if (useLegacyGuestFlow && !isGuest) {
+      return Promise.reject(
+        new Error('The user must be a guest to use the legacy flow'),
+      );
+    }
+
     const requestId = currentRequestId.current++;
     const actionMetadata = { userId, requestId, queryHash };
+    const getUserReturnsLegacyData =
```
Comment (boliveira) [inline packages/react/src/returns/hooks/useUserReturns.ts:186]: Shouldn't we validate if `useLegacyGuestFlow` is true but either `guestUserEmail` or `orderId` are undefined?

File: packages/react/src/returns/hooks/types/useUserReturns.ts (hunk 1 above)
Comment (boliveira) [reply packages/react/src/returns/hooks/types/useUserReturns.ts:10]: Actually scratch that, I see that you are using the `orderId` parameter from `GetUserReturnsQuery` which does not exist in `GetUserReturnsLegacyQuery`.

**Review by nelsonleite — APPROVED** [review]

File: packages/react/src/returns/hooks/useUserReturns.ts (hunk 2 above)
Comment (andreecosta) [reply packages/react/src/returns/hooks/useUserReturns.ts:195]: Added the validation. Please take a look.

**Review by jpklzm — APPROVED** [review]

**Review by boliveira — APPROVED** [review]

**Pull request merged**